packages/react-kit/src/components/Header/Header.tsx

```diff
@@ -0,0 +1,111 @@
+import { IconButton } from '@components/Button';
+import clsx from 'clsx';
+import FocusTrap from 'focus-trap-react';
+import React, { useEffect, useState } from 'react';
+import { CloseSVG, HamburgerMenuSVG } from './HeaderIcons';
+import { NavHeaderProps } from './props';
+
+export const Header = ({
+	Logo,
+	children,
+	overlayStyles,
+	navbarStyles,
+	focusTrapOptions,
+}: NavHeaderProps) => {
+	const [isOpen, setIsOpen] = useState(false);
+	const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+	const handleOpen = () => setIsOpen((prev) => !prev);
+
+	const MAX_MOBILE_NAV_WIDTH = 1023;
+
+	useEffect(() => {
+		const handleKeyUp = (e: KeyboardEvent) => {
+			if (e.key === 'Escape' || e.key === 'Esc') {
+				setIsOpen(false);
+			}
+		};
+		const handleResize = () => {
+			setWindowWidth(window.innerWidth);
+			windowWidth > MAX_MOBILE_NAV_WIDTH && setIsOpen(false);
+		};
+		setWindowWidth(window.innerWidth);
+
+		window.addEventListener('resize', handleResize);
+		window.addEventListener('keyup', handleKeyUp);
+
+		return () => {
+			window.removeEventListener('resize', handleResize);
+			window.removeEventListener('keyup', handleKeyUp);
+		};
+	}, [windowWidth]);
+
+	/**
+	 * Need to use a ref here so the FocusTrap can work properly.
+	 */
+	const NavList = React.forwardRef<
+		React.ElementRef<'nav'>,
+		{ children: React.ReactNode }
+	>(({ children }, ref) => {
+		return (
+			<nav
```

Comment:
That makes sense. I was struggling with that part of the design. Splitting it up would solve it nicely.

Reply:
I may need some more guidance here on the markup - are you envisioning something like this?

```tsx
<header>
	<Logo />
	<nav aria-label="Main">
		<ul>{mainNavItems}</ul>
	</nav>
	<nav aria-label="Secondary">
		<ul>{secondaryNavItems}</ul>
	</nav>
	...buttons
</header>;


```

Reply:
The buttons would be in the second nav, but otherwise this is what I was expecting. The actual markup I'm seeing in the latest branch looks right to me.